spec/acceptance/basic_dsc_resources/user/user_spec.rb

```diff
@@ -0,0 +1,219 @@
+    }
+
+    dsc_manifest = single_dsc_resource_manifest(dsc_type, dsc_props)
+
+    # Verify
+    error_msg = /Error:.*The specified username is invalid/m
+
+    windows_agents.each do | agent |
+      it 'Should apply the manifest and encounter an error' do
+        on(agent, puppet('apply'), :stdin => dsc_manifest, :acceptable_exit_codes => 0) do |result|
+          assert_match(error_msg, result.stderr, 'Expected error was not detected!')
+        end
+      end
+    end
+  end
+
+  context 'MODULES-2229 - C68752 - Apply DSC User Resource that Removes a User' do
+
+    # ERB Manifest
+    dsc_type = 'user'
+    dsc_module = 'PSDesiredStateConfiguration'
+    dsc_props = {
+      :dsc_ensure   => 'Present',
+      :dsc_username => 'catmandew'
+    }
+
+    dsc_manifest = single_dsc_resource_manifest(dsc_type, dsc_props)
+
+    windows_agents.each do | agent |
+      it 'Should apply the manifest successfully' do
+        on(agent, puppet('apply'), :stdin => dsc_manifest, :acceptable_exit_codes => [0,2]) do |result|
+          assert_no_match(/Error:/, result.stderr, 'Unexpected error was detected!')
+        end
+      end
+
+      it 'Should apply the manifest to remove the user' do
+        dsc_props[:dsc_ensure] = 'Absent'
+        dsc_remove_manifest = single_dsc_resource_manifest(dsc_type, dsc_props)
+        on(agent, puppet('apply'), :stdin => dsc_remove_manifest, :acceptable_exit_codes => [0,2]) do |result|
+          assert_no_match(/Error:/, result.stderr, 'Unexpected error was detected!')
+        end
+      end
+
+      it 'Should verify the results' do
+        assert_dsc_resource(
+          agent,
+          dsc_type,
+          dsc_module,
+          :Ensure   => 'Absent',
+          :UserName => dsc_props[:dsc_username]
+        )
+      end
+    end
+  end
+
+  context 'MODULES-2537 - C68751 - Apply DSC User Resource with Valid "UserName" and "Password" Specified' do
+
+    username = 'specialcat'
+    password = 'specialcatpassword'
+
+    # ERB Manifest
+    dsc_type = 'user'
+    dsc_module = 'PSDesiredStateConfiguration'
+    dsc_props = {
+      :dsc_ensure   => 'Present',
+      :dsc_username => username,
+      :dsc_password => "{'user' => '#{username}', 'password' => '#{password}'}"
+    }
+
+    dsc_manifest = single_dsc_resource_manifest(dsc_type, dsc_props)
+
+    windows_agents.each do | agent |
+      it 'Should apply a manifest with no errors' do
+        on(agent, puppet('apply'), :stdin => dsc_manifest, :acceptable_exit_codes => [0,2]) do |result|
+          assert_no_match(/Error:/, result.stderr, 'Unexpected error was detected!')
+        end
+      end
+
+      it 'Should verify the results' do
+        assert_dsc_cred_resource(
+          agent,
+          username,
+          password,
+          dsc_type,
+          dsc_module,
+          :Password,
+          :Ensure   => dsc_props[:dsc_ensure],
+          :UserName => username
+        )
+      end
+
+      it 'Should perform teardown' do
```

Comment:
typically teardown is done in an "after(:all)" block

Reply:
I agree and attempted to do, but the function definition for `assert_dsc_resource` actually contains an `assert` if you follow the definition down far enough. I can't put an example in `before` or `after` blocks. The assert means this is actually a test. Is a refactor for this in scope for this ticket? Because if it is, it also contains a redundant implementation of a function to execute powershell that I wouldn't mind getting rid of.

Reply:
Oh yeah, hmmm, I don't think we want to refactor that `assert_dsc_resource` method, but should scope that for future work.